source/ub.tex

```diff
@@ -23,6 +23,32 @@
 would result in the program having defined behavior. If no such set of objects would give the program defined
 behavior, the behavior of the program is undefined.
 
+\pnum
+\begin{example}
+\begin{codeblock}
+void f()
+{
+  void *p = malloc(sizeof(int) + sizeof(float));    // undefined behavior, cannot create
+                                                    // both int and float in same place
+  int& i = *reinterpret_cast<int*>(p);
+  float& f = *reinterpret_cast<float*>(p);
```

Comment:
Wouldn't this be the point of UB? 

CC @jensmaurer

Reply:
I wonder whether the behavior is still well-defined, as some `union { int n; float m; };` type can exist theoretically, and binding reference to an inactive member is probably OK.

Perhaps the following is better, as the latter assignment can't change the active member.
```C++
  *reinterpret_cast<int*>(p) = 0;
  *reinterpret_cast<float*>(p) = 0.0f;
```

(This also seems to indicate a kind of incompatibility with C. In C, assignment via a new type here can change the effective type.)

Reply:
Fixed.